hudi-client/src/main/java/org/apache/hudi/io/HoodieCommitArchiveLog.java

```diff
@@ -249,6 +249,13 @@ public void archive(List<HoodieInstant> instants) throws HoodieCommitException {
       LOG.info("Wrapper schema " + wrapperSchema.toString());
       List<IndexedRecord> records = new ArrayList<>();
       for (HoodieInstant hoodieInstant : instants) {
+
+        // filter empty instant, like *.commit.requested
+        byte[] instantDetails = commitTimeline.getInstantDetails(hoodieInstant).get();
```

Comment:
@lamber-ken : Instead of blindly skipping the empty metadata, can we handle in HoodieCommitArchiveLog.convertToAvroRecord so that we still create an archived entry but with empty commitMetadata. This way, we faithfully archive all state transitions. 

In other words, for the below two code snippets, if the state is Requested, skip setting commit metadata but set other attributes.
case HoodieTimeline.COMMIT_ACTION: {
        HoodieCommitMetadata commitMetadata = HoodieCommitMetadata
            .fromBytes(commitTimeline.getInstantDetails(hoodieInstant).get(), HoodieCommitMetadata.class);
        archivedMetaWrapper.setHoodieCommitMetadata(commitMetadataConverter(commitMetadata));
        archivedMetaWrapper.setActionType(ActionType.commit.name());
        break;
      }

case HoodieTimeline.DELTA_COMMIT_ACTION: {
        HoodieCommitMetadata commitMetadata = HoodieCommitMetadata
            .fromBytes(commitTimeline.getInstantDetails(hoodieInstant).get(), HoodieCommitMetadata.class);
        archivedMetaWrapper.setHoodieCommitMetadata(commitMetadataConverter(commitMetadata));
        archivedMetaWrapper.setActionType(ActionType.commit.name());
        break;
      }